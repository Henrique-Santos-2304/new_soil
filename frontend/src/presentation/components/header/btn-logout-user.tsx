import { ArrowBack } from '@mui/icons-material'
import { Button } from '@mui/material'

const BtnLogoutUser: React.FC = () => {
  return (
    <Button
      id="btn-logout"
      aria-label="botão para fazer logout na aplicação"
      variant="contained"
      startIcon={<ArrowBack />}
      sx={{ marginLeft: 3 }}
    >
      Sair
    </Button>
  )
}

export { BtnLogoutUser }
