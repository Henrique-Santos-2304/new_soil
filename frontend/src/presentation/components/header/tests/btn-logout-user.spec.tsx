import '@testing-library/jest-dom'
import { renderThemeMock } from '@/tests/index'
import { BtnLogoutUser } from '../btn-logout-user'

describe('Btn Logout Component', () => {
  it('should render component and write snapshot', () => {
    const { container } = renderThemeMock(<BtnLogoutUser />)
    expect(container).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should expected header to have button logout', () => {
    const { container } = renderThemeMock(<BtnLogoutUser />)

    const heading = container.querySelector('button[id="btn-logout"]')
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveProperty('id', 'btn-logout')
  })
})
