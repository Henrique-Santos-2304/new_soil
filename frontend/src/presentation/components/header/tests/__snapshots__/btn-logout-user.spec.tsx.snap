// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn Logout Component should render component and write snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-imz5j4"
  >
    <button
      aria-label="botão para fazer logout na aplicação"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1xri6ks-MuiButtonBase-root-MuiButton-root"
      id="btn-logout"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      Sair
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
